README.md

```diff
@@ -23,396 +23,345 @@ Stable release versions are manually released from the main branch.
+1. **Bitcoin Provider**: Handles Bitcoin transaction and message signing
+2. **Babylon Provider**: Handles Babylon transaction signing
 
-Using the Bitcoin staking scripts, you can generate a Bitcoin staking
-transaction and later sign it using a supported wallet's method.
-In this instance, we use the `btcWallet.signTransaction()` method.
+These providers are essential for constructing valid Babylon transaction 
+messages, as certain message fields depend on the output of signed Bitcoin 
+transactions. The `SigningStep` enum is used throughout the signing process to:
+- Identify which signing operation is being performed
+- Enable proper user interaction flows (e.g., showing relevant prompts)
 
 ```ts
-import { stakingTransaction } from "@babylonlabs-io/btc-staking-ts";
-import { Psbt, Transaction } from "bitcoinjs-lib";
-
-// stakingTransaction constructs an unsigned BTC Staking transaction
-const unsignedStakingPsbt: {psbt: Psbt, fee: number} = stakingTransaction(
-  scripts: {
-    timelockScript,
-    unbondingScript,
-    slashingScript,
-    dataEmbedScript
-  },
-  stakingAmount,
-  changeAddress,
-  inputUTXOs,
-  network(),
-  feeRate,
-  btcWallet.isTaproot ? btcWallet.publicKeyNoCoord() : undefined,
-  lockHeight,
-);
-
-const signedStakingPsbt = await btcWallet.signPsbt(unsignedStakingPsbt.psbt.toHex());
-const stakingTx = Psbt.fromHex(signedStakingPsbt).extractTransaction();
+export interface BtcProvider {
+  // Sign a PSBT
+  signPsbt(signingStep: SigningStep, psbtHex: string): Promise<string>;
+  // Sign a message using the ECDSA type
+  // This is optional and only required if you would like to use the 
+  // `createProofOfPossession` function
+  signMessage?: (
+    signingStep: SigningStep, message: string, type: "ecdsa" || "bip322-simple"
+  ) => Promise<string>;
+}
+
+export interface BabylonProvider {
+  signTransaction: <T extends object>(
+    signingStep: SigningStep,
+    msg: {
+      typeUrl: string;
+      value: T;
+    }
+  ) => Promise<Uint8Array>
+}
 ```
 
-Public key is needed only if the wallet is in Taproot mode, for `tapInternalKey`.
-
-### Create unbonding transaction
+### Post-Staking Registration
 
-The staking script allows users to on-demand unbond their locked stake before
-the staking transaction timelock expires, subject to an unbonding period.
+This refers to registration of an already existing BTC staking transaction into Babylon
+chain network. For more details, refer to [documentation](https://github.com/babylonlabs-io/babylon/blob/main/docs/register-bitcoin-stake.md#21-post-staking-registration).
 
-The unbonding transaction can be created as follows:
+> **Note**: This method does not create Bitcoin transactions that are ready to be sent to 
+> the Bitcoin network. Instead, it creates Babylon transactions that contain the
+> unsigned and partially signed Bitcoin transactions together with already 
+> onchain staking transaction required for delegation registration.
```

Comment:
What transactions are created? Which are signed and which are not? Does the user need to send anything to bitcoin?

I don't understand what information this paragraph communicates